### PR TITLE
Accept tabBar label to be a function

### PR DESCRIPTION
**Pull request by hose314**
It could be useful to dynamically change (`focused`) tabBar label default styles for certain tab(s).
This pull request is aim to discuss how to implement it properly. 

Usage example: 
```js
navigationOptions: {
  tabBar: {
    icon: ({ focused }) => <TabItem focused={focused} name="whatshot" activeColor="red" />,
    label: ({ focused, tintColor, defaultStyles }) => {
      return (<Text style={[defaultStyles, { color: focused ? 'red' : defaultStyles.color }]}>Profile</Text>);
    },
  }
},
```

Current implementation provided but without `Animated.Text` color specification

### CODE REVIEW

File: src/views/TabView/TabBarBottom.js [hunk 1]
```diff
@@ -89,6 +89,18 @@ export default class TabBarBottom extends PureComponent<DefaultProps, Props, voi
         </Animated.Text>
       );
     }
+    
+    if (typeof label === 'function') {
+      const st = [styles.label, { color: inactiveTintColor }, labelStyle];
+      const params = {
+        ...scene,
+        defaultStyles: StyleSheet.flatten(st)
```
Comment (satya164) [inline src/views/TabView/TabBarBottom.js]: Remove this. You can just return a string if you want to use default styles.

Comment (hose314) [inline src/views/TabView/TabBarBottom.js]: Sorry i don't get it, i want to make possible to leave all tabs with the same inactive styles but for certain tabs change it color when this tab is active. If i remove this i will lose information about `inactiveTintColor`

Comment (satya164) [inline src/views/TabView/TabBarBottom.js]: That sounds like very specific usecase. You already receive the `tintColor` as a prop, so you can write your own label component which animates between the colors when it changes. Also there's no reason to flatten the style here.

Comment (hose314) [inline src/views/TabView/TabBarBottom.js]: Agree. PR updated.

**Pull request merged**